tooltips: Fix tippy design where default tippy was used.

peterssonlinn opened the pull request:
<!-- Describe your pull request here.-->

Fixes #27817.

In several places an old design of tooltips were used. To create a more uniform UI, I have changed replaced several old tooltips to the tippy more commonly used. 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
| Before      | After |
| ----------- | ----------- |
|  <img width="456" alt="Screenshot 2023-11-27 at 11 07 57" src="https://github.com/zulip/zulip/assets/64125524/6ca3b9f4-099c-4138-9c80-763ee66794ea">| ![default_stream_settings](https://github.com/zulip/zulip/assets/64125524/a54dbaeb-d301-438e-877e-8ea63cb3ac1a)|
| <img width="753" alt="Screenshot 2023-11-27 at 11 06 58" src="https://github.com/zulip/zulip/assets/64125524/adde2eca-4c6a-4ed5-a1f3-6290816b59ff">  |![alert_word_settings](https://github.com/zulip/zulip/assets/64125524/9961cffa-f353-4790-acb3-2325ed7a384c)|
| <img width="810" alt="Screenshot 2023-11-27 at 11 08 26" src="https://github.com/zulip/zulip/assets/64125524/ea5d46b6-6e15-4790-a2ce-e6f135e2afbe">      | ![drafts_last_modified](https://github.com/zulip/zulip/assets/64125524/70f98e26-581c-4c23-9685-07f5a988f52b) |
| <img width="663" alt="Screenshot 2023-11-27 at 11 09 25" src="https://github.com/zulip/zulip/assets/64125524/31f1cf2d-32c8-4d34-9bf1-5e398fe6a026">      | ![preview_organization_profile](https://github.com/zulip/zulip/assets/64125524/3d1034fd-7a3e-4c5b-b657-b30bc45ee1e4)|
| <img width="282" alt="Screenshot 2023-11-27 at 11 09 49" src="https://github.com/zulip/zulip/assets/64125524/2d8b3239-d99a-419f-8751-c163df56195a">      | ![preview_profile_settings](https://github.com/zulip/zulip/assets/64125524/e51cdc67-49fa-45cf-896a-d3bab554f1c4)|
| <img width="732" alt="Screenshot 2023-11-27 at 11 17 45" src="https://github.com/zulip/zulip/assets/64125524/cb82e751-ab6e-47f4-adf3-83f5ade249ad">      | ![download_botserver](https://github.com/zulip/zulip/assets/64125524/19d55df4-0ecc-4001-959a-ee93dc6cabd3) |
| <img width="699" alt="Screenshot 2023-11-27 at 11 11 20" src="https://github.com/zulip/zulip/assets/64125524/1a34a83f-0cb5-42ca-99ab-ca3fc7abcf70">      | ![settings_custom_emoji](https://github.com/zulip/zulip/assets/64125524/35db2527-d83c-4ba3-bfc6-782afb5ec221)|
| <img width="732" alt="Screenshot 2023-11-27 at 11 12 08" src="https://github.com/zulip/zulip/assets/64125524/27663a7d-de74-4e18-90bc-7f60c592101e"> |![upload_file](https://github.com/zulip/zulip/assets/64125524/276038ee-c871-4a04-b131-9ad533869676) |
| <img width="699" alt="Screenshot 2023-11-27 at 11 10 28" src="https://github.com/zulip/zulip/assets/64125524/c8ffb4a5-17ae-431f-9c64-f0275f4e26f2"> | ![settings_linkfiers](https://github.com/zulip/zulip/assets/64125524/081a58c5-7e1a-46b5-895c-524318271e02)|
| <img width="538" alt="Screenshot 2023-11-27 at 09 29 45" src="https://github.com/zulip/zulip/assets/64125524/fe478b9f-5c92-4ad1-a8f5-cc3f9811d2df"> |![general_streams](https://github.com/zulip/zulip/assets/64125524/c6e7eb65-baef-43d1-bf32-f04fbb0c629b)|

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks! All the buttons where it's a pretty clear icon, but one not labeled with text, should have LONG_HOVER_DELAY. What questions does that leave?

alya commented:
Also, please pay attention to this comment:

> Also, we will not review/merge any changes to Settings/User Groups, as that panel will soon be deprecated.

peterssonlinn commented:
> Also, please pay attention to this comment:
> 
> > Also, we will not review/merge any changes to Settings/User Groups, as that panel will soon be deprecated.

Perfect! I will remove these ones.

Should I also remove the tippy under stream settings, or what do you think?

angelicaferlin commented:
Hi @peterssonlinn, I have read through your changes and think it looks good. The commit message is clear as well :)

alya commented:
> I found the following tooltip under stream settings. To me this does not create any more understanding of the stream, therefore I suggest that it should be removed. However, before doing so I would appreciate some response on it.

We should have a tooltip there, since the sometimes the stream name is too long to fit, and the tooltip offers a way to see the full name. EXTRA_LONG_HOVER_DELAY seems appropriate.

angelicaferlin commented:
@peterssonlinn I have checked your changes again and think it looks good :)

peterssonlinn commented:
Hi again @alya! Could you please assign someone to review this? I have rebased my branch so it shouldn't contain any conflicts.

peterssonlinn commented:
Hi @sayamsamal! I noticed that you were a reviewer on a similar PR as this one and were wondering if you could review this one as well?

peterssonlinn commented:
Hi @timabbott! I know you have a lot to work on with Zulip and I was wondering if you could let me know when this will be reviewed and who will do the review?

sayamsamal commented:
Adding an initial review over here, before I go ahead and review the code. I don't feel there is a need to show the tooltips over at "Preview organization profile" and "Preview profile" since it just repeats the same text as its reference element and doesn't add any additional information. We also don't have a breakpoint where we only show the icon, in which case having such a tooltip might be relevant.

Similarly, let's also remove the tooltip for the stream name over at the `Stream Settings > General`. We already have the stream name at two places, one on the header and one on the edit options.  It would have made sense if we only showed the truncated version of the stream name in the modal, but we do show the complete stream name on the top of the header.

sayamsamal commented:
@zulipbot add "maintainer review"

peterssonlinn commented:
> Adding an initial review over here, before I go ahead and review the code.
> 
> I don't feel there is a need to show the tooltips over at "Preview organization profile" and "Preview profile" since it just repeats the same text as its reference element and doesn't add any additional information. We also don't have a breakpoint where we only show the icon, in which case having such a tooltip might be relevant.
> 
> Similarly, let's also remove the tooltip for the stream name over at the `Stream Settings > General`. We already have the stream name at two places, one on the header and one on the edit options. It would have made sense if we only showed the truncated version of the stream name in the modal, but we do show the complete stream name on the top of the header.

Thanks for the comments! I have removed the tooltips that you mentioned and updated the resulting screen captures. Please comment if there is anything more that I should fix.

alya commented:
Nice!

Playing around with these, I think we should make all the "Delete X" tooltips just say "Delete". It's simpler, and it's pretty obvious what it is you'd be deleting. Similarly for "Edit X".

I think we can integrate this PR as-is, and do this as a follow-up. @peterssonlinn Are you interested in working on making that change throughout the settings panels?

timabbott commented on web/templates/draft.hbs:
```diff
@@ -16,7 +16,7 @@
                 </span>
                 <span class="recipient_bar_controls"></span>
-                <div class="recipient_row_date" title="{{t 'Last modified'}}">{{ time_stamp }}</div>
+                <div class="recipient_row_date restore-overlay-message tippy-zulip-delayed-tooltip" data-tooltip-template-id="last-modified-tooltip-template">{{time_stamp}}</div>
```
Why is `restore-overlay-message` added here?

timabbott replied:
Also, why use a tooltip template for this one? We usually only need those if there's HTML in the template, and this one should be able to just have the string "Last modified".

timabbott commented on web/templates/settings/uploaded_files_list.hbs:
```diff
@@ -20,17 +20,18 @@
-            <a type="submit" href="/user_uploads/{{path_id}}" class="btn no-style" title="{{t 'Download file' }}" id="download_attachment" download>
-                <i class="fa fa-download sea-green" aria-hidden="true"></i>
+            <a type="submit" href="/user_uploads/{{path_id}}" class="btn no-style" id="download_attachment" download>
+                <i class="fa fa-download sea-green tippy-zulip-delayed-tooltip" aria-hidden="true" data-tippy-content="{{t 'Download file' }}"></i>
```
Why is the tooltip moved from the outer `a` element ot the inner `i` element here? I don't see a strong reason to prevent one or the other.

timabbott commented on web/templates/settings/profile_settings.hbs:
```diff
@@ -63,7 +63,7 @@
             <button class="button rounded sea-green w-200 block" id="show_my_user_profile_modal">
                 {{t 'Preview profile' }}
-                <i class="fa fa-external-link" aria-hidden="true" title="{{t 'Preview profile' }}"></i>
+                <i class="fa fa-external-link"></i>
```
Hmm, should this have `aria-hidden` on it still? I would expect so giving it's not an interactable or readable element, just a visual decoration.

timabbott commented on web/templates/settings/organization_profile_admin.hbs:
```diff
@@ -51,7 +51,7 @@
         <a href="/login/?preview=true" target="_blank" rel="noopener noreferrer" class="button rounded sea-green w-200 block" id="id_org_profile_preview">
             {{t 'Preview organization profile' }}
-            <i class="fa fa-external-link" aria-hidden="true" title="{{t 'Preview organization profile' }}"></i>
+            <i class="fa fa-external-link"></i>
```
same ARIA question here.

timabbott commented on web/templates/settings/admin_linkifier_list.hbs:
```diff
@@ -14,10 +14,10 @@
             <i class="fa fa-pencil"></i>
         </button>
-        <button class="button small delete btn-danger" data-linkifier-id="{{id}}" title="{{t 'Delete' }}" aria-label="{{t 'Delete' }}">
+        <button class="button small delete btn-danger tippy-zulip-delayed-tooltip" data-linkifier-id="{{id}}" aria-label="{{t 'Delete' }}" data-tippy-content="{{t 'Delete' }}" >
```
I think the final space at the end (` >` ) isn't necessary or consistent with our usual style?

timabbott commented:
Thanks for all the work on this @peterssonlinn! I posted a full review; it'd be awesome if you can finish this off; if not, @sayamsamal can pick it up.

shashank-23002 commented:
Hey @peterssonlinn , as this PR was marked for completion candidate I would like to move this PR towards completion if you are currently not working on it. 

No worries if you are working . Just ping me once so that I can look for other issues.

Rinwaoluwa commented:
@peterssonlinn are you still working on this ?

amanagr commented:
Closing in favour of #28653
